### DISCUSSION TITLE
OS differences in displaying Hindi text

### DISCUSSION THREAD

**Discussion by pakkanen4** [ImageMagick/ImageMagick#7519] [Help]
### ImageMagick version

7.1.1-36 Q8

### Operating system, version and so on

AlmaLinux, Windows10

### Description

Hi, I have implemented a web app on Linux that has the functionality to generate images with Hindi text in ImageMagick (pango),
When I implement the same on Windows, the layouts are different.

As shown below, I executed a simple command.
```
magick -font "Noto Sans Devanagari Bold" pango:"<span size=\"19456\" background=\"green\">हिन्दी भाषा</span>" out.png
```

(Linux)
![hindi_linux](https://github.com/user-attachments/assets/f47fc043-9037-403e-b0ef-2d3639a90403)

(Windows)
![hindi_win](https://github.com/user-attachments/assets/65b242a9-6fd0-42f8-8631-fb15a8db7b52)

The generated images have different heights and top margins.

Is this a font issue or an OS issue?
Is there any way to make them match?

Thank you.

**Comment by fmw42** [comment]
Do you have the same versions of Pango?   Do you have the very same version of that font?

**Reply by pakkanen4** [reply]
Thanks, I didn't care the pango version because I hadn't explicitly specified.

I checked with 'dnf info pango' and it showed 1.48.7.
For windows (ImageMagick-7.1.1-36-portable-Q8-x86), the 'NOTICE.txt' file shows 1.45.3.

I don't know if this is the cause or not,
but I will try to set the pango version on the linux server to 1.45.3.

**Reply by pakkanen4** [reply]
As for fonts, I use NotoSansDevanagari-Bold.ttf from google fonts in both environments.

**Reply by pakkanen4** [reply]
It seemed difficult to build pango for me, so I changed the version of almalinux.

almalinux 9.3/pango 1.48.7 -> almalinux 8.8/pango 1.42.4

The result was the same as before the change.
It is unlikely that only 1.45.3 (between the above 2 versions) behaves differently.
